### PR TITLE
Feature/SOF-7326-1 feat: cleaned version of adding a test for JL

### PR DESCRIPTION
**Pull request by VsevolodX**

### CODE REVIEW

File: tests/cypress/e2e/menu/advanced/jupyterlite-transformation-dialog.feature [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    And I Run All Cells
+    And I see kernel status is Idle
+    And I submit materials
+    Then material with following data exists in state
+      | path              | index   |
+      | si-interface.json | $INT{2} |
+
+
+    # Reset the materials list
+    And I delete materials with index "2"
```
Comment (timurbazhirov) [inline tests/cypress/e2e/menu/advanced/jupyterlite-transformation-dialog.feature]: Newline

File: tests/cypress/support/commands.ts [hunk 2]
```diff
@@ -36,3 +39,87 @@
...
+                }
+                cy.log(`Until retry: "${result}" is not expected`);
+                cy.wait(delay);
+                return retry();
+            });
+        };
+
+        return retry();
+    },
+);
```
Comment (timurbazhirov) [inline tests/cypress/support/commands.ts]: Remove

File: tests/cypress/support/step_definitions/jupyterlite/I see kernel status is.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { When } from "@badeball/cypress-cucumber-preprocessor";
+
+import MaterialDesignerPage from "../../widgets/MaterialDesignerPage";
+
+When("I see kernel status is Idle", () => {
+    const { jupyterLiteSession } = new MaterialDesignerPage().designerWidget;
+
+    cy.until({
```
Comment (timurbazhirov) [inline tests/cypress/support/step_definitions/jupyterlite/I see kernel status is.ts]: Move to browser

File: tests/cypress/support/step_definitions/jupyterlite/I see kernel status is.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+        it: () => {
+            return jupyterLiteSession.isKernelIdle();
+        },
+        timeout: 100000,
```
Comment (timurbazhirov) [inline tests/cypress/support/step_definitions/jupyterlite/I see kernel status is.ts]: Timeouts should be in xs, sm, etc

File: tests/cypress/support/widgets/JupyterLiteSession.ts [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    restartKernel() {
+        this.browser.iframe(selectors.iframe).clickFirst(selectors.restartKernel, { force: true });
+        this.browser.iframe(selectors.iframe).waitForVisible(selectors.dialogAccept, "md");
+        this.browser.iframe(selectors.iframe).clickFirst(selectors.dialogAccept);
+    }
+
+    waitForKernelIdleWithRestart() {
+        // We need to wait some time to allow kernel to start on its own, if there's an issue, we restart and wait for some time again.
+        // Times are empirically determined.
+        cy.wait(15000);
```
Comment (timurbazhirov) [inline tests/cypress/support/widgets/JupyterLiteSession.ts]: We should not use waits, instead waitForVisible, wait for something to happen

File: tests/cypress/support/widgets/JupyterLiteTransformationDialogWidget.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+    selectMaterialsIn(index: number) {
+        this.browser.click(this.wrappedSelectors.materialsInSelector);
+        this.browser.click(selectors.materialsInSelectorItem(index));
+    }
+
+    verifyMaterialsOut(index: number) {
+        this.browser.click(this.wrappedSelectors.materialsOutSelector);
+        this.browser.waitForVisible(selectors.materialsOutSelectorItem(index), "xl");
+        // this.browser.click(selectors.materialsOutSelectorItem(index));
```
Comment (timurbazhirov) [inline tests/cypress/support/widgets/JupyterLiteTransformationDialogWidget.ts]: Remove

File: tests/cypress/support/step_definitions/jupyterlite/I see kernel status is.ts (hunk 3 above)
Comment (VsevolodX) [reply tests/cypress/support/step_definitions/jupyterlite/I see kernel status is.ts]: Here actually we don't need to await beyond timeout for `waitForVisible`

File: tests/cypress/support/widgets/JupyterLiteSession.ts [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+
+    waitForVisible() {
+        return this.iframeAnchor.waitForVisible(selectors.wrapper, "lg");
+    }
+
+    checkFileOpened(fileName: string) {
+        return this.iframeAnchor.waitForVisible(selectors.fileSelectorByFileName(fileName));
+    }
+
+    // TODO: change name to clickLinkInNotebookByItsTextContent
```
Comment (timurbazhirov) [inline tests/cypress/support/widgets/JupyterLiteSession.ts]: @VsevolodX to note the TODO items

**Review by timurbazhirov — APPROVED** [review]

**Pull request merged**